src/test/scala/io/sdkman/repos/VersionsRepoSpec.scala

```diff
@@ -85,6 +85,51 @@ class VersionsRepoSpec extends AnyWordSpec with Matchers with BeforeAndAfter wit
       }
     }
 
+    "attempt to find all versions by candidate and platform" when {
+
+      "that are set to be visible" in new TestRepo {
+        val java8u111 = Version("java", "8u111", "LINUX_64", "http://dl/8u111-b14/jdk-8u111-linux-x64.tar.gz", visible = Some(false))
+        val java8u121 = Version("java", "8u121", "LINUX_64", "http://dl/8u121-b14/jdk-8u121-linux-x64.tar.gz", visible = Some(true))
+
+        val javaVersions = Seq(java8u111, java8u121)
+
+        javaVersions.foreach(Mongo.insertVersion)
+
+        whenReady(findAllVersionsByCandidatePlatform("java", "LINUX_64")) { versions =>
```

Comment:
This function should probably be renamed to `findAllVisibleVersionsByCandidatePlatform` to express the true intent.

Reply:
Good idea!

Reply:
This will be a breaking change so we will bump the major release number for this.